crates/router/src/events.rs

```diff
@@ -82,9 +82,12 @@ impl EventsConfig {
 impl EventsHandler {
     pub fn log_event<T: KafkaMessage>(&self, event: &T) {
         match self {
-            Self::Kafka(kafka) => kafka.log_event(event).map_or((), |e| {
-                logger::error!("Failed to log event: {:?}", e);
-            }),
+            Self::Kafka(kafka) => kafka.log_event(event).map_or_else(
+                |e| {
+                    logger::error!("Failed to log event: {:?}", e);
+                },
+                |_| (),
+            ),
```

Comment:
```suggestion
            Self::Kafka(kafka) => kafka.log_event(event).unwrap_or_else(
                |e| {
                    logger::error!("Failed to log event: {:?}", e);
                }
            ),
```

In the current scenario we log an error for success state

Reply:
will need to use the `unwrap_or_else` function instead